components/merino_client/src/client.rs

```diff
@@ -79,19 +76,12 @@ impl MerinoClient {
         endpoint_url.query_pairs_mut().finish();
 
         let request = Request::get(endpoint_url);
-        let response: SuggestResponse = request
-            .send()
-            .map_err(|err| MerinoClientError::FetchFailed {
-                reason: err.to_string(),
-            })?
-            .require_success()
-            .map_err(|err| MerinoClientError::FetchFailed {
-                reason: err.to_string(),
-            })?
-            .json()
-            .map_err(|err| MerinoClientError::FetchFailed {
-                reason: err.to_string(),
-            })?;
+        let response = (|| -> Result<SuggestResponse, InternalError> {
```

Comment:
This is a pattern I learned from [kvstore](https://searchfox.org/mozilla-central/rev/81fbf5597044b6811e6aa4a10f3d78e8c3080f8c/toolkit/components/kvstore/src/task.rs#192-193,195,211)!